### DISCUSSION TITLE
Parallel Indexing

### DISCUSSION THREAD

**Discussion by niclashoyer** [OpenMage/magento-lts#2139] [Ideas]
### Description (*)
Magento 2 added parallel and "scoped" indexing in 2.2.6: https://community.magento.com/t5/Magento-DevBlog/Indexers-parallelization-and-optimization/ba-p/104922

Would it be possible to somehow add such a functionality to Magento 1? I did not have a deeper look into the indexation code, yet.

### Expected behavior (*)

Indexation should use parallelization where possible to speed up the process, e.g. if using multiple store views, run the index for every store view in parallel.

### Benefits
A lot of bigger Magento 1 installations could benefit from this, because indexation time depends not only on amount of products, but also on amount of store views.

**Comment by colinmollenhour** [comment]
As long as there is no licensing issue with the code I don't see why not, but just got to be careful there are no unwanted side-effects and hopefully the improvement justifies the work to port it.

**Comment by fballiano** [comment]
Do you have any code we could take a look at?

**Comment by Flyingmana** [comment]
There is also this project which was used by a few people successfully https://github.com/magento-hackathon/AsyncIndex

**Reply by empiricompany** [reply]
We also successfully use it with clients who have large catalogs. Would its license allow it to be ported to OpenMage?
